### DISCUSSION TITLE
How to set the label for an item in a list of objects.

### DISCUSSION THREAD

**Discussion by Yash-Singh1** [tinacms/tinacms#2840] [Help] [answered]
I have a field in one of my collections that looks something like this:

```ts
{
    name: 'projects',
    label: 'Projects',
    type: 'object',
    list: true,
    fields: [
        {
            name: 'name',
            label: 'Name',
            type: 'string',
        },
        {
            name: 'percentage_complete',
            label: 'Percentage Complete',
            type: 'number',
        },
        {
            name: 'priority',
            label: 'Priority',
            type: 'number',
        },
    ],
}
```

However, currently the list in the CMS editor looks like this:

![Screenshot from 2022-04-24 12-02-42](https://user-images.githubusercontent.com/53054099/164992300-8c168ea7-9715-4b73-8d3a-0dfa973fc0fa.png)

How am I supposed to change `Projects Item` to another label based on the data that I have present in the field when I am using a list of objects? I tried `itemProps`, but it doesn't seem to work and couldn't find any examples of people using it for solving this problem.

**Comment by coreyaus** [comment, marked as answer]
You're right that `itemProps` is the way to handle this. You can nest your `itemProps` configuration within the `ui` key, so it would look like this:

```javascript
{
    name: 'projects',
    label: 'Projects',
    type: 'object',
    list: true,
    ui: {
        itemProps: (item) => {
            return { label: item?.name };
        }
    },
    fields: [
        ...
    ]
}
```

You will also need to pass your `schema` into the `<TinaCMS>` component. Previously the `schema.ts` did not support functions for things like `itemProps` but the latest version does and the docs for that are a work in progress visible in this PR: https://github.com/tinacms/tinacms.org/pull/1293/files.

On earlier versions of `tinacms` it was still possible to use a custom `itemProps` function but you had to configure it via the `formifyCallback`. These recent updates allow you to define it directly inside the `ui` key within your `schema.ts` 😄 

Hope that helps!

**Reply by Yash-Singh1** [reply]
Thanks a lot!
